test/library/draft/arrow/Arrow.chpl

```diff
@@ -0,0 +1,253 @@
+module Arrow {
+    require "ArrowDecl.chpl";
+    public use ArrowDecl;
+    public use SysCTypes, CPtr;
+    // -------------------------- Type Declarations and Functions -------------
+
+    type ArrowArray = GArrowArray;
+
+    proc arrowInt32(): GArrowInt32DataType {
+        return garrow_int32_data_type_new();
+    }
+    
+    proc arrowInt64(): GArrowInt32DataType {
+        return garrow_int64_data_type_new();
+    }
+    proc arrowString(): GArrowStringDataType {
+        return garrow_string_data_type_new();
+    }
+    proc arrowBool(): GArrowBooleanDataType {
+        return garrow_boolean_data_type_new();
+    }
+    //--------------------------- Array building functions --------------------
+
+    proc array(array: [?arrDom] ?arrayType, validity: [?validityDom] int = [-1], 
+                specifiedType = nil): GArrowArray {
+        var fakeReturn: GArrowArray;
+        // Sorting based on the specifiedType and calling the appropriate function
+        //if(!specifiedType)
+        select arrayType {
+            when int do return int64Array(array, validity);
+            when string do return stringArray(array, validity);
+            when bool do return boolArray(array, validity);
+            otherwise {
+                writeln ("Unsupported type, \nreturning nil"); 
+                return fakeReturn;
+            }
+        }
+    }
+    
+    proc int64Array(array: [?arrDom] int, validity: [?validityDom] int) : GArrowInt64Array {
+        
+        var retval: GArrowInt64Array;
+        
+        var builder: GArrowInt64ArrayBuilder;
+        var success: gboolean = 1;
+        var error: GErrorPtr;
+
+        builder = garrow_int64_array_builder_new();
+        //writeln("0");
```

Comment:
Of course you know this, but if we want to actually get this merged, just a reminder to remove these comments.